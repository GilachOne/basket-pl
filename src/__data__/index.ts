import oopss from './animations/oopss.json';
import error from './animations/error.json';

export const animations = {
    oopss,
    error,
};
