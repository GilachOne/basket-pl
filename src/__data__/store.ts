import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';

import { api } from './api/main';
import { authApi } from './api/auth';
import { reducer as errorTextReducer } from './slices/errorTextSlice';

export const store = configureStore({
    reducer: {
        [api.reducerPath]: api.reducer,
        [authApi.reducerPath]: authApi.reducer,

        errorText: errorTextReducer,
    },

    middleware: getDefaultMiddleware =>
        getDefaultMiddleware()
            .concat(api.middleware)
            .concat(authApi.middleware),
});

setupListeners(store.dispatch);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
