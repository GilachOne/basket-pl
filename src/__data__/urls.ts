import { getNavigationsValue } from '@ijl/cli';

const baseURL = getNavigationsValue('basket.main');

export const URLs = {
    landing: {
        url: `${baseURL}`,
        isEnabled: getNavigationsValue('basket.main'),
    },
    dashboard: {
        url: `${baseURL}${getNavigationsValue('basket.dashboard')}`,
        isEnabled: getNavigationsValue('basket.dashboard'),
    },
    signIn: {
        url: `${baseURL}${getNavigationsValue('basket.sign.in')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.in'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.in')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.in'),
    },
    signUp: {
        url: `${baseURL}${getNavigationsValue('basket.sign.up')}`,
        getUrlWidthEmail: email => {
            if (email) {
                return `${baseURL}${getNavigationsValue(
                    'basket.sign.up'
                )}?email=${email}`;
            }
            return `${baseURL}${getNavigationsValue('basket.sign.up')}`;
        },
        isEnabled: getNavigationsValue('basket.sign.up'),
    },
    list: {
        url: `${baseURL}${getNavigationsValue('basket.list')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.list')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.list'),
    },
    add: {
        url: `${baseURL}${getNavigationsValue('basket.add')}`,
        getUrl: listId =>
            `${baseURL}${getNavigationsValue('basket.add')}`.replace(
                ':id',
                listId
            ),
        isEnabled: getNavigationsValue('basket.add'),
    },
    resetPassword: {
        url: `${baseURL}${getNavigationsValue('basket.reset.password')}`,
        isEnabled: getNavigationsValue('basket.reset.password'),
    },
};
