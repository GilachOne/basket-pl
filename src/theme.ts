export const theme = {
    colors: {
        brand: {
            light2: '#08DA36',
            light: '#74D14C',
            main: '#74D14C',
            dark: '#54AB26',
        },
        background: {
            light: '#FFFFFF',
			main: '#008060',
        },
        accent: {
            light6: '#CCCCCC',
            light5: '#F6EAFF',
            light4: '#d2d8e0',
            light3: '#FAF9F8',
            light2: '#D2D5D9',
            light: '#F6EAFF',
            main: '#9828F0',
            dark: '#2C107A',
        },
        text: {
            light3: '#A68FE5',
            light2: '#42474C',
            light: '#6B7177',
            main: '#FFFFFF',
            dark: '#000000',
        },
        bg: {
            main: '#000000',
        },
        error: {
            main: '#FF0000',
        },
    },
};
