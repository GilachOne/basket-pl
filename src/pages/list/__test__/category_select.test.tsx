import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { describe, it, expect, beforeAll, afterAll } from '@jest/globals';
import { render, waitFor, screen, fireEvent } from '@testing-library/react';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import 'whatwg-fetch';
import { ThemeProvider } from '@emotion/react';

import CategorySelect from '../../list/category_select';
import { theme } from '../../../theme';
import { store } from '../../../__data__/store';
import currentCategories from '../../../../stubs/json/list/categories/current/success.json';
import addCategories from '../../../../stubs/json/list/categories/add/success.json';

const server = setupServer(
    rest.get('/api/categories', (req, res, ctx) => {
        return res(ctx.json(currentCategories));
    }),
    rest.post('/api/categories', (req, res, ctx) => {
        return res(ctx.json(addCategories));
    })
);

describe('testing CategorySelect', () => {
    beforeAll(() => {
        server.listen();
    });
    afterAll(() => {
        server.close();
    });

    it('test render CategorySelect Loading', () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <CategorySelect />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });

    it('test render CategorySelect current categories', async () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <CategorySelect />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        await waitFor(() => screen.getByText('Продукты'));
        expect(container).toMatchSnapshot();
    });

    it('test render CategorySelect add categories', async () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <CategorySelect />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        await waitFor(() => screen.getByText('Продукты'));
		
		fireEvent.click(screen.getByTitle('Кнопка для добавления категорий'));
		fireEvent.click(screen.getByTitle('Добавить категорию'));
        expect(container).toMatchSnapshot();

        fireEvent.change(
            screen.getByPlaceholderText('Введите название категории'),
            { target: { value: '%овощи' } }
        );
        fireEvent.click(screen.getByTitle('Добавить категорию'));
        expect(container).toMatchSnapshot();

		fireEvent.change(
            screen.getByPlaceholderText('Введите название категории'),
            { target: { value: 'Фрукты' } }
        );
        fireEvent.change(screen.getByTestId('select'), {
            target: { value: '#FCF20E' },
        });
        fireEvent.click(screen.getByTitle('Добавить категорию'));
        expect(container).toMatchSnapshot();

        fireEvent.change(
            screen.getByPlaceholderText('Введите название категории'),
            { target: { value: 'Овощи' } }
        );
        fireEvent.change(screen.getByTestId('select'), {
            target: { value: '#B11F1F' },
        });
        fireEvent.click(screen.getByTitle('Добавить категорию'));
		await waitFor(() => screen.getByTitle('Кнопка для добавления категорий'));
        expect(container).toMatchSnapshot();
    });

    it('test render CategorySelect add Item', async () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <CategorySelect />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        await waitFor(() => screen.getByText('Продукты'));
		fireEvent.change(screen.getByPlaceholderText('Введите текст'), {
            target: { value: 'Кефир' },
        });
		fireEvent.click(screen.getByTitle('Добавить'));
        expect(container).toMatchSnapshot();
        
        fireEvent.change(screen.getByPlaceholderText('Введите текст'), {
            target: { value: 'Молоко' },
        });
        fireEvent.click(screen.getByText('Продукты'));
		fireEvent.click(screen.getByTitle('Добавить'));
        expect(container).toMatchSnapshot();

		fireEvent.change(screen.getByPlaceholderText('Введите текст'), {
            target: { value: 'Хлеб_' },
        });
        fireEvent.click(screen.getByText('Продукты'));
		fireEvent.click(screen.getByTitle('Добавить'));
        expect(container).toMatchSnapshot();
    });

	it('test render CategorySelect add Item span error', async () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <CategorySelect />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        await waitFor(() => screen.getByText('Продукты'));
		screen.getByPlaceholderText('Введите текст').focus();
        fireEvent.click(screen.getByTitle('Добавить'));
		await waitFor(() => screen.getByText('Необходимо выбрать категорию покупки'));
        expect(container).toMatchSnapshot();
	});

    it('test render CategorySelect click button back', async () => {
        const { container } = render(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <CategorySelect />
                    </BrowserRouter>
                </ThemeProvider>
            </Provider>
        );
        await waitFor(() => screen.getByText('Продукты'));

        fireEvent.click(screen.getByTitle('Назад'));
        expect(container).toMatchSnapshot();
    });
});