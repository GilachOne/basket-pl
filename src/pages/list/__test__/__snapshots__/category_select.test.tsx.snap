// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CategorySelect test render CategorySelect Loading 1`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-6i4ykg"
    >
      <span
        class="css-1m6qh8b"
      />
      <span
        class="css-1fni309"
      />
      <span
        class="css-oqbc0s"
      />
      <span
        class="css-1olkxn8"
      />
      <span
        class="css-1vc4txl"
      />
      <span
        class="css-hkkqum"
      />
      <span
        class="css-1nfmhdq"
      />
      <span
        class="css-1jyf7or"
      />
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add Item 1`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          >
            Кефир
          </textarea>
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add Item 2`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-sbcdrl"
            color="#D2D5D9"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            disabled=""
            name="text"
            placeholder="Введите текст"
          >
            Молоко
          </textarea>
        </div>
        <button
          class="css-10uh9e6"
          disabled=""
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add Item 3`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-sbcdrl"
            color="#D2D5D9"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            disabled=""
            name="text"
            placeholder="Введите текст"
          >
            Хлеб_
          </textarea>
          <span
            class="css-uexi7o"
          >
            Специальные символы запрещены, кроме символа: процента, запятой, номера, тире
          </span>
        </div>
        <button
          class="css-10uh9e6"
          disabled=""
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add Item span error 1`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <span
          class="css-uexi7o"
        >
          Необходимо выбрать категорию покупки
        </span>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
          <span
            class="css-uexi7o"
          >
            Название покупки не может быть пустым
          </span>
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add categories 1`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        />
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
      <form>
        <p
          class="css-1enixo5"
        >
          Если нет подходящей категории:
        </p>
        <div
          class="css-4ok39k"
        >
          <input
            class="css-1w4q98z"
            name="name"
            placeholder="Введите название категории"
            type="text"
            value=""
          />
          <span
            class="css-uexi7o"
          >
            Название категории не может быть пустой
          </span>
          <select
            class="css-178zj4d"
            data-testid="select"
            name="color"
          >
            <option
              disabled=""
              value=""
            >
              Выберите цвет
            </option>
            <option
              value="#08AE0F"
            >
              Зеленый
            </option>
            <option
              value="#9D79B9"
            >
              Лиловый
            </option>
            <option
              value="#B11F1F"
            >
              Алый
            </option>
            <option
              value="#3414F5"
            >
              Синий
            </option>
            <option
              value="#008060"
            >
              Тёмно-зелёный
            </option>
            <option
              value="#D928D2"
            >
              Розовый
            </option>
            <option
              value="#9B763D"
            >
              Коричневый
            </option>
            <option
              value="#830DF9"
            >
              Фиолетовый
            </option>
            <option
              value="#FA8803"
            >
              Оранжевый
            </option>
            <option
              value="#FCF20E"
            >
              Жёлтый
            </option>
            <option
              value="#EC1818"
            >
              Красный
            </option>
            <option
              value="#14F52B"
            >
              Светло-зеленый
            </option>
          </select>
          <span
            class="css-uexi7o"
          >
            Необходимо выберать цвет категории
          </span>
        </div>
        <button
          class="css-15jozr4"
          title="Добавить категорию"
          type="submit"
        >
          Добавить категорию
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add categories 2`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        />
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
      <form>
        <p
          class="css-1enixo5"
        >
          Если нет подходящей категории:
        </p>
        <div
          class="css-4ok39k"
        >
          <input
            class="css-1w4q98z"
            name="name"
            placeholder="Введите название категории"
            type="text"
            value="%овощи"
          />
          <span
            class="css-uexi7o"
          >
            Название категории должно содержать только буквы, допустимо тире
          </span>
          <select
            class="css-178zj4d"
            data-testid="select"
            name="color"
          >
            <option
              disabled=""
              value=""
            >
              Выберите цвет
            </option>
            <option
              value="#08AE0F"
            >
              Зеленый
            </option>
            <option
              value="#9D79B9"
            >
              Лиловый
            </option>
            <option
              value="#B11F1F"
            >
              Алый
            </option>
            <option
              value="#3414F5"
            >
              Синий
            </option>
            <option
              value="#008060"
            >
              Тёмно-зелёный
            </option>
            <option
              value="#D928D2"
            >
              Розовый
            </option>
            <option
              value="#9B763D"
            >
              Коричневый
            </option>
            <option
              value="#830DF9"
            >
              Фиолетовый
            </option>
            <option
              value="#FA8803"
            >
              Оранжевый
            </option>
            <option
              value="#FCF20E"
            >
              Жёлтый
            </option>
            <option
              value="#EC1818"
            >
              Красный
            </option>
            <option
              value="#14F52B"
            >
              Светло-зеленый
            </option>
          </select>
          <span
            class="css-uexi7o"
          >
            Необходимо выберать цвет категории
          </span>
        </div>
        <button
          class="css-15jozr4"
          title="Добавить категорию"
          type="submit"
        >
          Добавить категорию
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add categories 3`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        />
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
      <form>
        <p
          class="css-1enixo5"
        >
          Если нет подходящей категории:
        </p>
        <div
          class="css-4ok39k"
        >
          <input
            class="css-1w4q98z"
            disabled=""
            name="name"
            placeholder="Введите название категории"
            type="text"
            value="Фрукты"
          />
          <select
            class="css-178zj4d"
            data-testid="select"
            disabled=""
            name="color"
          >
            <option
              disabled=""
              value=""
            >
              Выберите цвет
            </option>
            <option
              value="#08AE0F"
            >
              Зеленый
            </option>
            <option
              value="#9D79B9"
            >
              Лиловый
            </option>
            <option
              value="#B11F1F"
            >
              Алый
            </option>
            <option
              value="#3414F5"
            >
              Синий
            </option>
            <option
              value="#008060"
            >
              Тёмно-зелёный
            </option>
            <option
              value="#D928D2"
            >
              Розовый
            </option>
            <option
              value="#9B763D"
            >
              Коричневый
            </option>
            <option
              value="#830DF9"
            >
              Фиолетовый
            </option>
            <option
              value="#FA8803"
            >
              Оранжевый
            </option>
            <option
              value="#FCF20E"
            >
              Жёлтый
            </option>
            <option
              value="#EC1818"
            >
              Красный
            </option>
            <option
              value="#14F52B"
            >
              Светло-зеленый
            </option>
          </select>
        </div>
        <button
          class="css-15jozr4"
          disabled=""
          title="Добавить категорию"
          type="submit"
        >
          Добавить категорию
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect add categories 4`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect click button back 1`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`testing CategorySelect test render CategorySelect current categories 1`] = `
<div>
  <div>
    <div
      class="css-10m13ut"
    >
      <div
        class="css-1gwlyb6"
      >
        <a
          class="css-12eo6k0"
          href="/url/url/"
          title="Назад"
        >
          <img
            alt="back"
            src="arrow-left.svg"
          />
        </a>
      </div>
    </div>
    <div
      class="css-1gwlyb6"
    >
      <ul
        class="css-1loydxm"
      >
        <li
          class="css-kagisi"
        >
          <h2
            class="css-156op5v"
          >
            Выберите категорию:
          </h2>
        </li>
        <li
          class="css-kagisi"
        >
          <button
            class="css-kxqetq"
            title="Кнопка для добавления категорий"
            type="button"
          >
            +
          </button>
        </li>
      </ul>
      <form>
        <div
          class="css-139b3em"
        >
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Продукты"
            type="button"
          >
            <div
              class="css-1q4bltv"
              color="#08AE0F"
            />
            Продукты
          </button>
          <button
            class="css-p42m1b"
            color="#FFFFFF"
            title="Бытовая химия"
            type="button"
          >
            <div
              class="css-1ewgiqb"
              color="#3414F5"
            />
            Бытовая химия
          </button>
        </div>
        <p
          class="css-1enixo5"
        >
          Введите название покупки:
        </p>
        <div
          class="css-dw8up7"
        >
          <textarea
            class="css-14sbxw9"
            name="text"
            placeholder="Введите текст"
          />
        </div>
        <button
          class="css-10uh9e6"
          title="Добавить"
          type="submit"
        >
          Добавить
        </button>
      </form>
    </div>
  </div>
</div>
`;
